src/foam/nanos/pm/PM.js

```diff
@@ -203,24 +250,47 @@ foam.CLASS({
               this(fo.getClassInfo(), name);
             }
 
+            public PM(String... args) {
+              if ( args.length > 0 ) {
+                setKey(args[0]);
+              }
+              if ( args.length > 1 ) {
+                setName(combine(java.util.Arrays.copyOfRange(args, 1, args.length)));
+              }
+              init_();
+            }
+
             public PM(Object... args) {
-              setKey(args[0].toString());
-              StringBuilder sb = new StringBuilder();
-              for (Object obj: java.util.Arrays.copyOfRange(args, 1, args.length)){
-                sb.append(obj.toString()).append(":");
+              if ( args.length > 0 ) {
+                setKey(args[0].toString());
+              }
+              if ( args.length > 1 ) {
+                setName(combine(java.util.Arrays.copyOfRange(args, 1, args.length)));
               }
-              if ( sb.length() > 0 )
-                setName(sb.deleteCharAt(sb.length() - 1).toString());
               init_();
             }
 
             private static String combine(String... args) {
+              if ( args == null ) {
+                return "";
+              }
               StringBuilder sb = new StringBuilder();
               for ( String s: args) {
                 sb.append(s).append(":");
               }
               return sb.deleteCharAt(sb.length() - 1).toString();
             }
+
+            private static String combine(Object... args) {
+              if ( args == null ) {
+                return "";
+              }
+              StringBuilder sb = new StringBuilder();
+              for ( Object o: args) {
+                sb.append(o.toString()).append(":");
```

Comment:
Is .toString() needed? Then, do we need a String... version if we already have an Object... version?